proto/cosmos/base/tendermint/v1beta1/query.proto

```diff
@@ -136,3 +141,47 @@ message Module {
   // checksum
   string sum = 3;
 }
+
+// ABCIQueryRequest defines the request structure for the ABCIQuery gRPC query.
+message ABCIQueryRequest {
+  bytes  data   = 1;
+  string path   = 2;
+  int64  height = 3;
+  bool   prove  = 4;
+}
+
+// ABCIQueryResponse defines the response structure for the ABCIQuery gRPC query.
+//
+// Note: This type is a duplicate of the ResponseQuery proto type defined in
+// Tendermint.
+message ABCIQueryResponse {
+  uint32 code = 1;
+  // DEPRECATED: use "value" instead
+  // bytes data = 2;
+  string   log       = 3;  // nondeterministic
+  string   info      = 4;  // nondeterministic
+  int64    index     = 5;
+  bytes    key       = 6;
+  bytes    value     = 7;
+  ProofOps proof_ops = 8;
+  int64    height    = 9;
+  string   codespace = 10;
+}
+
+// ProofOp defines an operation used for calculating Merkle root. The data could
+// be arbitrary format, providing nessecary data for example neighbouring node
+// hash.
+//
+// Note: This type is a duplicate of the ProofOp proto type defined in Tendermint.
```

Comment:
Why are we duplicating?

Reply:
https://github.com/cosmos/cosmos-sdk/pull/11017#pullrequestreview-861418564

Reply:
I just realized we have `proto/tendermint/abci/types.proto`. Should we use those types instead? Is that what you meant?

Reply:
I remember there was something about not using TM types, but can't remember what (maybe @marbar3778 knows?). I'm okay to keep duplicated.

Reply:
For now all the TM proto files are actually duplicated in the SDK, but we are still using their generated types. We would like the TM team to actually version their proto files and following buf breaking change rules so there aren't incompatible breaking changes propagated up to us. But for now I think it's just okay to re-use and deal with that later